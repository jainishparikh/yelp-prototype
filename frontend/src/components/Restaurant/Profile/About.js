import React, { Component } from 'react'
import { Redirect } from 'react-router';
import { Link } from 'react-router-dom'
import cookie from 'react-cookies';
import Modal from 'react-modal';
import axios from 'axios';
import AddDish from '../Dishes/AddDishes';
import IndividualDish from '../Dishes/IndividualDish';
import BACKEND_URL from '../../../config/config';
import profile_picture from '../../../images/restaurant.jpeg'

export class RestaurantAbout extends Component {
    constructor( props ) {
        super( props )
        this.state = {
            restaurantID: "",
            name: "",
            email: "",
            contact: "",
            location: "",
            description: "",
            timing: "",
            dishPopUp: false,
            profileImagePath: profile_picture,
            dishes: []
        }

    }
    componentDidMount () {
        let email = cookie.load( "email" )
        axios.get( BACKEND_URL + '/restaurants/about/' + email ).then( ( response ) => {
            if ( response.status === 200 ) {
                let imagePath = BACKEND_URL + "/images/profilepics/" + response.data.profilePicture
                if ( response.data.profilePicture === null || response.data.profilePicture === "" ) {
                    console.log( "inside imagepath null" )
                    imagePath = profile_picture
                }
                this.setState( {
                    restaurantID: response.data._id,
                    name: response.data.name,
                    email: response.data.email,
                    contact: response.data.contact,
                    location: response.data.location,
                    description: response.data.description,
                    timing: response.data.timing,
                    profileImagePath: imagePath,
                    dishes: response.data.dishes,
                } )
            }

        } ).catch( ( err ) => {
            console.log( " error getting restaurant data" )
            this.setState( {
                error: true
            } )

        } );
    }

    toggleDishPopUp = ( e ) => {
        this.setState( {
            dishPopUp: !this.state.dishPopUp
        } )
    }


    render () {
        var redirectVar = null;
        if ( !( cookie.load( "auth" ) && cookie.load( "type" ) === "restaurants" ) ) {
            redirectVar = <Redirect to="/login" />
        }

        let displayDishImages = this.state.dishes.map( ( dish ) => {

            var dishImagePath = BACKEND_URL + "/images/dishes/" + dish.dishPicture
            return (
                <img src={ dishImagePath } style={ { "margin": "10px" } } width="200px" height="90%" alt="" />
            )
        } )

        let details = this.state.dishes.map( ( dish ) => {
            return (
                <IndividualDish key={ dish.dishID } dishData={ dish } />
            )
        } )

        return (
            < div >
                { redirectVar }
                <div className="container-fluid m-1" style={ { height: "100vh" } }>
                    <div className="row h-100">
                        <div className="col-3" style={ { "borderRight": "1px solid #e6e6e6", "background": "whitesmoke" } }>
                            <div className="profile-info" style={ { height: "80%" } }>
                                <table style={ { height: "100%" } }>
                                    <tbody>
                                        <div className="restaurant-info" style={ { height: "60%" } }>
                                            <tr> <img src={ this.state.profileImagePath } width="102%" height="100%" alt="" /></tr>
                                            <tr><h2>{ this.state.name }</h2></tr>
                                            <tr>{ this.state.location }</tr>
                                            <tr>{ this.state.description }</tr>
                                        </div>
                                        <div className="rstaurant-contact" style={ { height: "40%" } }>
                                            <tr><h5>Contact Detals:</h5></tr>
                                            <tr>{ this.state.contact }</tr>
                                            <tr>{ this.state.email }</tr>
                                            <tr>{ this.state.timing }</tr>
                                        </div>
                                    </tbody>
                                </table>
                            </div>
                            <div className="profile-edit" style={ { height: "20%" } }>
                                <Link className="btn btn-primary" to={ {
                                    pathname: "/restaurants/editprofile", state: {
                                        userData: this.state
                                    }
                                } }>Edit Profile</Link>
                            </div>
                        </div>
                        <div className="col-9" style={ { "paddingLeft": "40px" } }>
                            <div className="row">
                                <h3>Here's What We Offer</h3>
                            </div>
                            <div className="row">
                                <div style={ {
                                    "width": "auto",
                                    "overflow": "auto",
                                    "whiteSpace": "nowrap",
                                    "height": "200px"
                                } }>
                                    { displayDishImages }
                                </div>
                            </div>
                            {/* Add dish */ }
                            <div className="row mt-3 mb-3">

                                <div className="add-dish m-2" >
                                    <button className="btn btn-danger" onClick={ this.toggleDishPopUp }>Add Dish</button>
                                </div>
                                {/* using react-modal for popup to add dish */ }
                                <Modal isOpen={ this.state.dishPopUp } style={ {
                                    overlay: {
                                        position: 'fixed',
                                        top: 0,
                                        left: 0,
                                        right: 0,
                                        bottom: 0,
                                        backgroundColor: 'rgba(255, 255, 255, 0.25)'
                                    },
                                    content: {
                                        position: 'relative',
                                        top: '20%',
                                        left: '20%',
                                        right: '20%',
                                        bottom: '20%',
                                        border: '2px solid #ccc',
                                        background: '#fff',
                                        overflow: 'auto',
                                        WebkitOverflowScrolling: 'touch',
                                        borderRadius: '4px',
                                        outline: 'none',
                                        padding: '20px'
                                    }
                                } } >
                                    <AddDish call="add" closePopUp={ this.toggleDishPopUp } />
                                </Modal>



                            </div>
                            {/* Display dishes */ }
                            <div className="row" >
                                <div className="dishes">
                                    { details }
                                </div>
                            </div>

                        </div>

                    </div>
                </div>


            </div >
        )

    }

}

export default RestaurantAbout
