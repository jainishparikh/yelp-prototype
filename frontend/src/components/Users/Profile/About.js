import React, { Children, Component } from 'react'
import { Redirect } from 'react-router';
import { Link } from 'react-router-dom'
import cookie from 'react-cookies';
import axios from 'axios';
import profile_picture from '../../../images/profile.png';
import BACKEND_URL from '../../../config/config';
import GetReviews from '../Reviews/GetReviews';
import getUserProfileAction from '../../../actions/getUserProfileAction'
import { connect } from "react-redux";

export class UserAbout extends Component {
    constructor( props ) {
        super( props )
        this.state = {
            userID: "",
            name: "",
            nickName: "",
            email: "",
            contactNumber: "",
            dateOfBirth: "",
            city: "",
            state: "",
            country: "",
            headline: "",
            yelpingSince: "",
            thingsILove: "",
            blogLink: "",
            profileImagePath: profile_picture,
        }

    }
    componentDidMount () {

        this.props.getUserProfileAction( this.state );
        // let email = cookie.load( "email" )
        // axios.defaults.headers.common[ "authorization" ] = cookie.load( 'token' )
        // axios.defaults.withCredentials = true;
        // return axios.get( BACKEND_URL + '/users/about/' + email ).then( ( response ) => {
        //     // console.log( response )
        //     if ( response.status === 200 ) {
        //         console.log( "got data", response.data._id )
        //         let imagePath = BACKEND_URL + "/images/profilepics/" + response.data.profilePicture
        //         if ( response.data.profilePicture === null || response.data.profilePicture === "" ) {
        //             console.log( "inside imagepath null" )
        //             imagePath = profile_picture
        //         }
        //         this.setState( {
        //             userID: response.data._id,
        //             name: response.data.name,
        //             nickName: response.data.nickName,
        //             email: response.data.email,
        //             contactNumber: response.data.contactNumber,
        //             dateOfBirth: response.data.dateOfBirth,
        //             city: response.data.city,
        //             state: response.data.state,
        //             country: response.data.country,
        //             headline: response.data.headline,
        //             yelpingSince: response.data.yelpingSince,
        //             thingsILove: response.data.thingsILove,
        //             blogLink: response.data.blogLink,
        //             profileImagePath: imagePath
        //         } )
        //     }
        //     // console.log( this.state );

        // } ).catch( ( err ) => {
        //     console.log( " error getting user data", err )
        //     this.setState( {
        //         error: true
        //     } )

        // } );
    }



    render () {
        var redirectVar = null;
        if ( !( cookie.load( "auth" ) && cookie.load( "type" ) === "users" ) ) {
            redirectVar = <Redirect to="/login" />
        }

        return (

            < div >
                { redirectVar }
                <div className="container-fluid">
                    <div className="container-fluid" style={ { height: "100vh" } }>
                        <div className="row mt-3 mb-3" style={ { height: "40%" } }>
                            {/* profile picture */ }
                            <div className="col-3">
                                <img src={ this.props.profileImagePath } style={ { "box-shadow": "0px 0px 10px gray" } } width="300px" height="300px" alt="" />
                            </div>
                            {/* profile display */ }
                            <div className="col-8" >
                                <div className="row pt-4" style={ { "background": "whitesmoke", "box-shadow": "0px 0px 10px gray" } }>
                                    <div className="col-8">
                                        <ul style={ { "list-style-type": "none" } }>
                                            <li><h2>{ this.props.user.name }</h2></li>
                                            <li>&nbsp;</li>
                                            <li><h5>{ this.props.user.city } { this.props.user.state }</h5></li>
                                            <li>{ this.props.user.email }&nbsp;</li>
                                            <li>&nbsp;</li>
                                            <li>&nbsp;</li>
                                            <li>{ this.props.user.headline }&nbsp;</li>
                                        </ul>

                                    </div>

                                    <div className="col-4" style={ { "padding": "0 15px", "border-left": "1px solid #e6e6e6" } }>
                                        <ul style={ { "list-style-type": "none" } }>
                                            <li><h5>Yelping Since:</h5></li>
                                            <li>{ this.props.user.yelpingSince }&nbsp;</li>
                                            <li>&nbsp;</li>
                                            <li><h5>Things I Love:</h5></li>
                                            <li>{ this.props.user.thingsILove }&nbsp;</li>
                                            <li>&nbsp;</li>
                                            <li><h5>Blog Link:</h5></li>
                                            <li>{ this.props.user.blogLink }&nbsp;</li>
                                        </ul>

                                    </div>
                                </div>

                            </div>


                        </div>

                        <div className="row" style={ { height: "60%" } }>
                            {/* edit-profile */ }
                            <div className="col-3" style={ { "marginTop": "20px" } } >
                                <Link className="btn btn-primary" to={ {
                                    pathname: "/users/editprofile", state: {
                                        userData: this.props.user
                                    }
                                } }>

                                    Edit Profile
                                </Link>
                            </div>
                            {/* reviews */ }
                            <div className="col-8" style={ { "padding": "0 15px", "border-left": "1px solid #e6e6e6" } }>
                                <GetReviews reviewData={ this.state } />

                            </div>


                        </div>
                    </div>


                </div >
            </div >
        )
    }

}

const matchStateToProps = ( state ) => {
    return {
        user: state.getUserProfileReducer.userData,
        profileImagePath: state.getUserProfileReducer.profileImagePath,
    }

}

const matchDispatchToProps = ( dispatch ) => {
    return {
        getUserProfileAction: ( data ) => dispatch( getUserProfileAction( data ) ),
    }
}

export default connect( matchStateToProps, matchDispatchToProps )( UserAbout )