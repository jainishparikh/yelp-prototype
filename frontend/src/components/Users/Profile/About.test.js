import React from 'react';
import { shallow } from 'enzyme';

import About from './About'

describe( 'About', () => {

    let axios = require( "axios" );
    let MockAdapter = require( "axios-mock-adapter" );
    let mock = new MockAdapter( axios );

    const userData =
    {
        userID: 8,
        name: "Goku",
        nickName: "DBZ",
        email: "user2@gmail.com",
        contactNumber: "3333333333",
        dateOfBirth: "2000-01-01",
        city: "San Jose",
        state: "California",
        country: "USA",
        headline: "I love Indian Cuisine",
        yelpingSince: "Sept 2020",
        thingsILove: "",
        blogLink: "",
        profilePicture: "myimage123.jpeg"

    }



    mock.onGet( "http://localhost:3001/users/about/user2@gmail.com" ).reply( 200,
        userData
    );


    let component

    beforeEach( () => {
        cookie.setCookie( "email", "user2@gmail.com" )
        component = shallow( <About debug /> );
    } );

    it( 'should render correctly in "debug" mode', () => {
        component.instance()
        expect( component ).toMatchSnapshot();
    } );

    it( 'validate cokkie has values', () => {
        expect( cookie.load( "email" ) ).toEqual( "user2@gmail.com" )
    } );

    it( 'check if userdata was recieved and matches', async ( done ) => {
        component
            .instance()
            .componentDidMount()
            .then( () => {
                expect( component.state().name ).toEqual( userData.name )
                expect( component.state().contactNumber ).toEqual( userData.contactNumber )
                done()
            } )
        expect( component ).toMatchSnapshot();
    } );

    // it( 'simulate checkbox to filter restaurants by curbside_pickup', async ( done ) => {
    //     component
    //         .instance()
    //         .componentDidMount()
    //         .then( () => {
    //             let restaurants_filter = component.find( 'input[type="checkbox"]' ).at( 0 )
    //             expect( component.state().filtered_restaurants ).toEqual( restaurants )
    //             restaurants_filter.simulate( 'change', { target: { checked: true, value: "curbside_pickup" } } )
    //             expect( component.state().filtered ).toEqual( "curbside_pickup" )
    //             expect( component.state().filtered_restaurants ).toEqual( filtered_restaurants )
    //             restaurants_filter.simulate( 'change', { target: { checked: false, value: "curbside_pickup" } } )
    //             expect( component.state().filtered ).toEqual( "" )
    //             expect( component.state().filtered_restaurants ).toEqual( restaurants )
    //             done()
    //         } )

    //     expect( component ).toMatchSnapshot();
    // } )

    // it( 'simulate search restaurants to search indian cuisine', async ( done ) => {
    //     component
    //         .instance()
    //         .componentDidMount()
    //         .then( () => {
    //             let search_options = component.find( 'select.searchOptions' )
    //             let search = component.find( 'input[type="text"]' )
    //             expect( component.state().filtered_restaurants ).toEqual( restaurants )
    //             search_options.simulate( 'keydown', { keyCode: 40 } )
    //             search_options.simulate( 'keydown', { keyCode: 13 } )
    //             search_options.simulate( 'change', { target: { name: "selectedOption", value: "cuisine" } } )
    //             expect( component.state().selectedOption ).toEqual( "cuisine" )
    //             search.simulate( 'change', { target: { name: "search", value: "indian" } } )
    //             expect( component.state().search ).toEqual( "indian" )
    //             component
    //                 .instance()
    //                 .searchRestaurants( {
    //                     preventDefault: () => {
    //                     }
    //                 } )
    //                 .then( () => {
    //                     expect( component.state().filtered_restaurants ).toEqual( filtered_restaurants )
    //                     done()
    //                 } )

    //             expect( component ).toMatchSnapshot();
    //         } )
    // } )

    // it( 'validate localstorage is empty', () => {
    //     localStorage.removeItem( "id" )
    //     localStorage.removeItem( "active" )
    //     expect( localStorage.store ).toEqual( {} )
    // } );

} )