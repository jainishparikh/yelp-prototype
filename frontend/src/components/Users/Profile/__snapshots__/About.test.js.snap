// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About check if userdata was recieved and matches 1`] = `ShallowWrapper {}`;

exports[`About should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
